packages/teams-js/src/private/externalAppCardActions.ts

```diff
@@ -127,9 +127,14 @@ export namespace externalAppCardActions {
    * Limited to Microsoft-internal use
    * @param appId ID of the application the request is intended for. This must be a UUID
    * @param url The URL to open
+   * @param fromElement The element on behalf of which the M365 app is making the request
    * @returns Promise that resolves to ActionOpenUrlType indicating the type of URL that was opened on success and rejects with ActionOpenUrlError if the request fails
    */
-  export function processActionOpenUrl(appId: string, url: URL): Promise<ActionOpenUrlType> {
+  export function processActionOpenUrl(
+    appId: string,
+    url: URL,
+    fromElement?: { name: 'composeExtensions' | 'plugins' },
```

Comment:
>? [](http://example.com/codeflow?start=15&length=1)

Is there a reason that this has to be optional? Can't we just make the app that needs this always specify?

At minimum we should document the default behavior when this isn't specified.

Reply:
We have to make it optional because M365 cannot send the info just yet. On hostSDK side, Hosts are not ready to consume a new parameter yet either. Adding an optional will not break them. It would let them consume the new parameter when they are ready.

Hosts have a dependency on element type, and they need to do things differently based on composeExtension vs plugin. Right now, they can handle just composeExtensions.  Even if they could handle plugins today, sending a default value won't be beneficial because they would not be able to trust the value and execute the request.

Reply:
fromElement will be useful only when M365 sends the value. 

Tagging Erin @erinha and Saurabh @sbtron

Reply:
Does that mean that this parameter will eventually be made required?

Can you please document the behavior in the description of the `fromElement` parameter on line 130 for what happens if a consumer doesn't provide a value?

Reply:
Does this accurately describe the logic?

If `fromElement` is not provided, the information from the manifest is used to determine whether the URL can be processed by the host. Deep link URLs for plugins are not supported and will result in an error.